### PR TITLE
Data View Window Startup helpful Information

### LINKED ISSUE #5632: More helpful information at start-up of R-Instat
This suggestion is from a new user, Michael Clapham.

When you start R-Instat it has the message "No data loaded" on the left-hand window and a window waiting for results on the right.

Michael commented that it wasn't clear to him where he might start.  He compared with Excel that would start with a blank sheet - so does the old Instat.

Could we have a more informative message on what the user could do first?  Usually they will either use 
File > New Data Frame, or 
File > Open or 
File > Open from Library.  

Perhaps there might be a picture too?

### PR DESCRIPTION
**Pull request by Patowhiz**
Fixes #5632.
@africanmathsinitiative/developers this is ready for review.

### CODE REVIEW

**Comment by rdstern** [comment]
This looks really good.  
a) In the top section add the ... after New Data Frame
b) Make the space much wider to show the filenames as much as possible in the middle section
c) The More doesn't work.  I suspect there are only 5 files.  If that takes a long time to implement, then delete the option for now.
d) In the Help the first item could be the `Introduction`, so it goes to the top of the Help, i.e. as in Help > Introduction.  The second item could be `R-packages in R-Instat`.  This links to Help > R Packages and Commands.  The third item could be R-Instat website and link to `http://r-instat.org/ `(Why isn't it https?).  The 4th item could be `Instructional videos (to come)` and is not live.  The fifth could then be `(more to come)` and is also not live.

**Review by rdstern — CHANGES_REQUESTED** [review]
I showed it to Rachel who was very happy with the new screen.
a) It still needs to be much wider for the Recent file names.
b) The More as you showed it with the other software, then opened another scrollable window.  Yours just adds the file names below.  I liked the idea of the other window, but (for now) I suggest you delete the More, unless that can be done quickly.  Or keep it, but disable.

**Comment by Patowhiz** [comment]
@rdstern I've made the changes requested in your above comment. File names and file paths are now fully viewable.
Will the videos be online? Do we have page template already online, I could link to it. I have the videos link disabled for now.



@africanmathsinitiative/developers this is now ready for review.
@lloyddewit this PR has changes to frmMain.vb.

**Comment by lloyddewit** [comment]
@Ivanluv please could you peer review? thanks

**Review by rdstern — DISMISSED** [review]
@Patowhiz I am now very happy with this improvement to the start of R-Instat.  I note that it was not what Danny and I suggested, but, after discussion and suggestions from Patrick,  it is considerably better!
Last time I asked for changes in the More button and really like what there is now.  I had not realised that there would be scroll bars.

**Comment by Ivanluv** [comment]
@lloyddewit  this looks okay from my end

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@Patowhiz  This looks very good. I really liked your commenting approach. I just have a couple of small comments. Please could you check them asap so we can merge this today and include in the new release?

File: instat/clsRecentFiles.vb [hunk 1]
```diff
@@ -24,7 +24,13 @@ Public Class clsRecentFiles
...
     Private sepEnd As ToolStripSeparator
     ' declare a variable to contain the most recent opened items
-    Private strListMRU As New List(Of String)
+    Private strListMRU As New List(Of String)
+
+    ''' <summary>
+    ''' stores reference to the  the data view window used in frmMain. 
+    ''' will be used to add recent files menu items to to it.
+    ''' </summary>
+    Private ucrDataViewWindow As ucrDataView
```
[the remark this answers is not in the document]
Comment (Patowhiz) [reply instat/clsRecentFiles.vb]: @lloyddewit A property will expose this control to other classes that may use this this class in future. I'm not sure if that is desired?When I went through this class, it seemed like a utility class to me. I'm happy to make this a property if you strongly feel there is a merit to this.

[the remark this answers is not in the document]
Comment (lloyddewit) [reply instat/clsRecentFiles.vb]: @Patowhiz Thank you for the explanation. I'm happy to leave the code as it is. I was just curious if you had a special reason for doing it this way.
I understand your concerns about this class needing to know about the user control. I think having a private data member with a public `SetDataViewWindow` function is similar to a public property in terms of write access, it's mainly the synatx that differs.

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern Please could you re-approve (no functional changes since your last approval)? thanks

**Review by rdstern — APPROVED** [review]

File: instat/ucrDataView.Designer.vb [hunk 2]
```diff
@@ -530,41 +540,211 @@ Partial Class ucrDataView
...
+        Me.linkHelpMoreToCome.AutoSize = True
+        Me.linkHelpMoreToCome.Enabled = False
+        Me.linkHelpMoreToCome.ImeMode = System.Windows.Forms.ImeMode.NoControl
+        Me.linkHelpMoreToCome.LinkBehavior = System.Windows.Forms.LinkBehavior.NeverUnderline
+        Me.linkHelpMoreToCome.Location = New System.Drawing.Point(30, 409)
+        Me.linkHelpMoreToCome.Name = "linkHelpMoreToCome"
+        Me.linkHelpMoreToCome.Size = New System.Drawing.Size(72, 13)
+        Me.linkHelpMoreToCome.TabIndex = 11
+        Me.linkHelpMoreToCome.TabStop = True
+        Me.linkHelpMoreToCome.Text = "More to come"
```
Comment (dannyparsons) [inline instat/ucrDataView.Designer.vb:595]: Can we hide this or remove this? I don't think this is useful if it doesn't do anything.

File: instat/ucrDataView.Designer.vb [hunk 3]
```diff
@@ -530,41 +540,211 @@ Partial Class ucrDataView
...
+        Me.linkHelpInstructionVideos.AutoSize = True
+        Me.linkHelpInstructionVideos.Enabled = False
+        Me.linkHelpInstructionVideos.ImeMode = System.Windows.Forms.ImeMode.NoControl
+        Me.linkHelpInstructionVideos.LinkBehavior = System.Windows.Forms.LinkBehavior.NeverUnderline
+        Me.linkHelpInstructionVideos.Location = New System.Drawing.Point(30, 390)
+        Me.linkHelpInstructionVideos.Name = "linkHelpInstructionVideos"
+        Me.linkHelpInstructionVideos.Size = New System.Drawing.Size(98, 13)
+        Me.linkHelpInstructionVideos.TabIndex = 10
+        Me.linkHelpInstructionVideos.TabStop = True
+        Me.linkHelpInstructionVideos.Text = "Instructional videos"
```
Comment (dannyparsons) [inline instat/ucrDataView.Designer.vb:609]: Also hide this until there is a link

File: instat/ucrDataView.Designer.vb [hunk 4]
```diff
@@ -530,41 +540,211 @@ Partial Class ucrDataView
...
+        Me.linkHelpRpackages.ActiveLinkColor = System.Drawing.Color.Blue
+        Me.linkHelpRpackages.AutoSize = True
+        Me.linkHelpRpackages.ImeMode = System.Windows.Forms.ImeMode.NoControl
+        Me.linkHelpRpackages.LinkBehavior = System.Windows.Forms.LinkBehavior.NeverUnderline
+        Me.linkHelpRpackages.Location = New System.Drawing.Point(30, 352)
+        Me.linkHelpRpackages.Name = "linkHelpRpackages"
+        Me.linkHelpRpackages.Size = New System.Drawing.Size(116, 13)
+        Me.linkHelpRpackages.TabIndex = 8
+        Me.linkHelpRpackages.TabStop = True
+        Me.linkHelpRpackages.Text = "R-packages in R-Instat"
```
Comment (dannyparsons) [inline instat/ucrDataView.Designer.vb:635]: R packages instead of R-packages

Comment (Patowhiz) [reply instat/ucrDataView.Designer.vb:635]: Done this in PR #6036

File: instat/ucrDataView.Designer.vb (hunk 3 above)
Comment (Patowhiz) [reply instat/ucrDataView.Designer.vb:609]: Done this in PR #6036

File: instat/ucrDataView.Designer.vb (hunk 2 above)
Comment (Patowhiz) [reply instat/ucrDataView.Designer.vb:595]: Done this in PR #6036

**Comment by Patowhiz** [comment]
I've made the changes in #6036

**Pull request merged**